### PR TITLE
Patrick Lazo

### PR DESCRIPTION
**Pull request by patrick100**
All exercises solved.

### CODE REVIEW

File: src/answers.sql [hunk 1]
```diff
@@ -1,15 +1,125 @@
 -- Your answers here:
 -- 1
+SELECT
+    c.name,
+    count(s.name)
+FROM
+    countries c
+    RIGHT JOIN states s ON c.id = s.country_id
```
Comment (ederivero) [inline src/answers.sql]: Why a right join and not a left join or inner join? 🤔

File: src/answers.sql [hunk 2]
```diff
@@ -1,15 +1,125 @@
...
+            o.country_id,
+            address,
+            count(o.id)
+        FROM
+            offices o
+            INNER JOIN employees e ON o.id = e.office_id
+        GROUP BY
+            o.id
+    ) t
+    INNER JOIN countries c ON t.country_id = c.id
```
Comment (ederivero) [inline src/answers.sql]: it's more efficient if you use an inner join instead of a subquery.

File: src/answers.sql [hunk 3]
```diff
@@ -1,15 +1,125 @@
...
+    count DESC
+LIMIT
+    3;
 
 -- 5
+SELECT
+    count(*) AS list_of_office
+FROM
+    states s
+    INNER JOIN offices o ON s.id = o.state_id
```
Comment (ederivero) [inline src/answers.sql:55]: What happens if we have 2 states _Colorado_ in different countries?

File: src/answers.sql [hunk 4]
```diff
@@ -1,15 +1,125 @@
...
+-- 8
+SELECT
+    e.uuid,
+    CONCAT(e.first_name, ' ', e.last_name) AS full_name,
+    e.email,
+    e.job_title,
+    o.name AS company,
+    c.name AS country,
+    s.name AS state,
+    e2.first_name AS boss_name
```
Comment (ederivero) [inline src/answers.sql]: for more readability instead of `e` and `e2` you can use e.g. `employee` and `boss` alias

File: src/answers.sql (hunk 1 above)
Comment (patrick100) [reply src/answers.sql]: Yes, my bad I misunderstood the question I tried to count the states that don't have a country.

File: src/answers.sql (hunk 2 above)
Comment (patrick100) [reply src/answers.sql]: Ok, I will fix it.

File: src/answers.sql (hunk 3 above)
Comment (patrick100) [reply src/answers.sql:55]: It will don't work.

File: src/answers.sql (hunk 4 above)
Comment (patrick100) [reply src/answers.sql]: ok.

**Pull request closed without merging**